packet.go

```diff
@@ -13,8 +13,15 @@ type Extension struct {
 	payload []byte
 }
 
+func NewExtension(id uint8, payload []byte) Extension {
+	return Extension{
```

Comment:
Considering the functions receiver is *Extension, shouldn't you return a pointer?

Reply:
And I wonder, did have any impact?